src/app/application/components/data-source/DataSource.vue

```diff
@@ -1,5 +1,15 @@
-<script setup lang="ts">
+<template>
+  <slot
+    :data="(message as any)"
```

Comment:
I'm sorry but I'm going to be tedious for the first couple of ones :(

why is this `any`? can't we have a concrete type there? or does it not give us anything?

Reply:
Note that this bit was just moved around from the bottom of the file (all our components start with the template).

Typing this component is hard because what we effectively need is _the component_ (i.e. not some JavaScript function) to be typed with a generic parameter. Then, `data` should be something like `DataSource<T>` instead of `any`.

Vue (as of _very_ recently) supports typing components as generics, but when I tried it locally, I couldn’t avoid some TypeScript compilation errors that were introduced by doing so.

In short, doing this better is currently not possible.